libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -330,6 +374,23 @@ ReturnCode SoftwareContainer::shutdownGateways()
 
 std::shared_ptr<ContainerAbstractInterface> SoftwareContainer::getContainer()
 {
+    std::string id = std::string(m_container->id());
+
+    if (m_containerState == ContainerState::INVALID) {
+        std::string message = "Container is invalid " + id;
+        log_error() << message;
+        throw InvalidContainerError(message);
+    }
+
+    if (m_containerState != ContainerState::READY
+        && m_containerState != ContainerState::SUSPENDED)
```

Comment:
Does it make sense to allow access to container implementation when it is suspended? What can you possibly want to do except resuming/destroying it?